src/GraphQL/loaders/parseClassTypes.js

```diff
@@ -341,10 +235,14 @@ const load = (
     name: classGraphQLConstraintTypeName,
     description: `The ${classGraphQLConstraintTypeName} input type is used in operations that involve filtering objects by a pointer field to ${graphQLClassName} class.`,
     fields: {
-      _eq: defaultGraphQLTypes._eq(classGraphQLScalarType),
-      _ne: defaultGraphQLTypes._ne(classGraphQLScalarType),
-      _in: defaultGraphQLTypes._in(classGraphQLScalarType),
-      _nin: defaultGraphQLTypes._nin(classGraphQLScalarType),
+      _eq: defaultGraphQLTypes._eq(GraphQLID),
+      _ne: defaultGraphQLTypes._ne(GraphQLID),
+      _in: defaultGraphQLTypes._in(
+        new GraphQLList(defaultGraphQLTypes.OBJECT_ID)
+      ),
+      _nin: defaultGraphQLTypes._nin(
+        new GraphQLList(defaultGraphQLTypes.OBJECT_ID)
+      ),
       _exists: defaultGraphQLTypes._exists,
       _select: defaultGraphQLTypes._select,
```

Comment:
I think we should also improve these following 4 operators. But we can address it in a separate PR. I will take a note in the project.